libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -1031,11 +1031,21 @@ method rpcHandler*(g: GossipSub,
 
     g.mcache.put(msgId, msg)
 
-    if g.verifySignature and not msg.verify(peer.peerId):
+    if (msg.signature.len > 0 or g.verifySignature) and not msg.verify(peer.peerId):
```

Comment:
`peer.peerId` is not really needed for verify, not sure why it's there..?

Reply:
It's indeed not used